### PR TITLE
Tree Distances I Java Solution

### PR DESCRIPTION
**Pull request by brad182**

### CODE REVIEW

File: solutions/cses-1132.mdx [hunk 1]
```diff
@@ -80,7 +83,98 @@ int main() {
...
+</Warning>
+	
+```java
+import java.util.*;
+import java.io.*;
+
+public class TreeDistancesI {
+	// source -> https://cses.fi/problemset/task/1132/
+
+	static ArrayList<Integer>[] adj;
```
Comment (Aadit-Ambadkar) [inline solutions/cses-1132.mdx:102]:
```suggestion
// source -> https://cses.fi/problemset/task/1132/

import java.util.*;
import java.io.*;

public class TreeDistancesI {
	static ArrayList<Integer>[] adj;
```

File: solutions/cses-1132.mdx [hunk 2]
```diff
@@ -80,7 +83,98 @@ int main() {
...
+	static int [] d1 = new int[200001];  // to store first-max length.
+	static int [] d2 = new int[200001]; // to store second-max length.
+	static int [] c = new int[200001]; // to store child for path of max length.
+
+	public static void dfs(int v, int p) {
+		d1[v] = 0;
+		d2[v] = 0;
+		for (int x : adj[v]) {
+			if (x == p)continue;
```
Comment (Aadit-Ambadkar) [inline solutions/cses-1132.mdx:111]:
```suggestion
			if (x == p) continue;
```

File: solutions/cses-1132.mdx [hunk 3]
```diff
@@ -80,7 +83,98 @@ int main() {
...
+import java.util.*;
+import java.io.*;
+
+public class TreeDistancesI {
+	// source -> https://cses.fi/problemset/task/1132/
+
+	static ArrayList<Integer>[] adj;
+	static int [] d1 = new int[200001];  // to store first-max length.
+	static int [] d2 = new int[200001]; // to store second-max length.
+	static int [] c = new int[200001]; // to store child for path of max length.
```
Comment (Aadit-Ambadkar) [inline solutions/cses-1132.mdx:105]: 85 characters using indents as 8 spaces, 81 characters using indents as 4 spaces. Either way, above 80 character limit.

**Review by Aadit-Ambadkar — CHANGES_REQUESTED** [review]

File: solutions/cses-1132.mdx [hunk 4]
```diff
@@ -80,7 +83,98 @@ int main() {
...
+<Warning>
+Java exceeds the time limit on two test cases.
+</Warning>
+	
+```java
+import java.util.*;
+import java.io.*;
+
+public class TreeDistancesI {
+	// source -> https://cses.fi/problemset/task/1132/
```
Comment (nathangong) [inline solutions/cses-1132.mdx:100]: Do you need this? It's just a link to the problem

Comment (Aadit-Ambadkar) [reply solutions/cses-1132.mdx:100]: It is to keep consistent with the java solution.

File: solutions/cses-1132.mdx (hunk 3 above)
Comment (nathangong) [inline solutions/cses-1132.mdx:105]: Can you make more descriptive variable names so it's easier to understand the rest of the code?

File: solutions/cses-1132.mdx [hunk 5]
```diff
@@ -80,7 +83,98 @@ int main() {
...
+
+	static ArrayList<Integer>[] adj;
+	static int [] d1 = new int[200001];  // to store first-max length.
+	static int [] d2 = new int[200001]; // to store second-max length.
+	static int [] c = new int[200001]; // to store child for path of max length.
+
+	public static void dfs(int v, int p) {
```
Comment (nathangong) [inline solutions/cses-1132.mdx:107]: Comments explaining what each dfs method does would be nice

File: solutions/cses-1132.mdx [hunk 6]
```diff
@@ -80,7 +83,98 @@ int main() {
...
+```java
+import java.util.*;
+import java.io.*;
+
+public class TreeDistancesI {
+	// source -> https://cses.fi/problemset/task/1132/
+
+	static ArrayList<Integer>[] adj;
+	static int [] d1 = new int[200001];  // to store first-max length.
+	static int [] d2 = new int[200001]; // to store second-max length.
```
Comment (dutinmeow) [inline solutions/cses-1132.mdx:104]: Two spaces before comment

File: solutions/cses-1132.mdx [hunk 7]
```diff
@@ -80,7 +83,98 @@ int main() {
 	}
 	return 0;
 }
+```				 
+</CPPSection>
+	
+<JavaSection>
+	
+<Warning>
+Java exceeds the time limit on two test cases.
```
Comment (dutinmeow) [inline solutions/cses-1132.mdx:92]: ... . This is unavoidable because Java has an inherently slow speed compared to C++ and other languages

**Review by dutinmeow — CHANGES_REQUESTED** [review]
pretty good

File: solutions/cses-1132.mdx (hunk 4 above)
Comment (bqi343) [reply solutions/cses-1132.mdx:100]: Then remove both occurrences, there's no reason to include it when the link to the problem already appears on the solution page.

**Review by bqi343 — CHANGES_REQUESTED** [review]

File: solutions/cses-1132.mdx (hunk 4 above)
Comment (andrewwangva) [reply solutions/cses-1132.mdx:100]: ^Agree with ben

File: solutions/cses-1132.mdx [hunk 8]
```diff
@@ -80,7 +83,98 @@ int main() {
...
+		}
+
+		for (int i = 0; i < n - 1; i++) {
+			u = io.nextInt();
+			v = io.nextInt();
+			adj[u].add(v);
+			adj[v].add(u);
+		}
+
+		// METHOD-I- All longest paths-given in CPH ~~~ (O(n))- 2-DFS
```
Comment (andrewwangva) [inline solutions/cses-1132.mdx:162]: Method 1?

**Comment by SansPapyrus683** [comment]
lmao i'll pile on after all the other reviews have been addressed

File: solutions/cses-1132.mdx (hunk 3 above)
Comment (brad182) [reply solutions/cses-1132.mdx:105]: its not a strict limit 81-85 is fine

File: solutions/cses-1132.mdx (hunk 7 above)
Comment (brad182) [reply solutions/cses-1132.mdx:90]: people already assume this. this is the same format as other java warnings

File: solutions/cses-1132.mdx [hunk 9]
```diff
@@ -80,7 +81,97 @@ int main() {
...
+		d2[v] = 0;
+		for (int x : adj[v]) {
+			if (x == p) continue;
+			dfs(x, v);
+			if (d1[x] + 1 > d1[v]) {
+				d2[v] = d1[v];
+				d1[v] = d1[x] + 1;
+				c[v] = x;
+			}
+			else if (d1[x] + 1 > d2[v]) {
```
Comment (Aadit-Ambadkar) [inline solutions/cses-1132.mdx:115]:
```suggestion
			} else if (d1[x] + 1 > d2[v]) {
```

File: solutions/cses-1132.mdx [hunk 10]
```diff
@@ -80,7 +81,97 @@ int main() {
...
+	public static void dfs2(int v, int p) {
+		for (int x : adj[v]) {
+			if (x == p) continue;
+			if (c[v] == x) {
+				if (d1[x] < d2[v] + 1) {
+					d2[x] = d1[x];
+					d1[x] = d2[v] + 1;
+					c[x] = v;
+				}
+				else {
```
Comment (Aadit-Ambadkar) [inline solutions/cses-1132.mdx:130]:
```suggestion
				} else {
```

File: solutions/cses-1132.mdx [hunk 11]
```diff
@@ -80,7 +81,97 @@ int main() {
...
+					d2[x] = Math.max(d2[x], d2[v] + 1);
+				}
+			}
+			else {
```
Comment (Aadit-Ambadkar) [inline solutions/cses-1132.mdx:134]:
```suggestion
			} else {
```

**Review by Aadit-Ambadkar — CHANGES_REQUESTED** [review]

File: solutions/cses-1132.mdx (hunk 9 above)
Comment (brad182) [reply solutions/cses-1132.mdx:115]: this qualifies as an optional change

Comment (Aadit-Ambadkar) [reply solutions/cses-1132.mdx:114]: Wait no it doesn't, it's not a personal preference. Oracle's styling guide dictates that you nest the if statements like above.

Comment (SansPapyrus683) [reply solutions/cses-1132.mdx:114]: also who the heck likes it that way anyways

File: solutions/cses-1132.mdx (hunk 3 above)
Comment (Aadit-Ambadkar) [reply solutions/cses-1132.mdx:101]: image

Comment (SansPapyrus683) [reply solutions/cses-1132.mdx:101]: aadit i don't know if you've been living under a rock but literally everyone else feels that 81-85 is fine- as long as it doesn't wrap on the website it's fine

File: solutions/cses-1132.mdx (hunk 9 above)
Comment (dutinmeow) [reply solutions/cses-1132.mdx:114]: +1, this is taking formatting a bit too far

File: solutions/cses-1132.mdx (hunk 3 above)
Comment (dutinmeow) [reply solutions/cses-1132.mdx:101]: make comment spacing consistant

Comment (dutinmeow) [reply solutions/cses-1132.mdx:101]: pls not another sans 1 is enough

**Review by dutinmeow — APPROVED** [review]
Make the changes mentioned above

**Review by bqi343 — APPROVED** [review]
My comment was addressed (haven’t reviewed the rest though.)

File: solutions/cses-1132.mdx [hunk 12]
```diff
@@ -2,60 +2,65 @@
...
 #define FIO ios::sync_with_stdio(0); cin.tie(0);
 
 
 vector<int> adj[200001];
-int d1[200001];  // to store first-max length.
-int d2[200001]; // to store second-max length.
-int c[200001]; // to store child for path of max length.
+int firstMax[200001];  // to store first-max length.
+int secondMax[200001];  // to store second-max length.
+int c[200001];   // to store child for path of max length.
```
Comment (SansPapyrus683) [inline solutions/cses-1132.mdx:25]: store 2e5 in a MAX_N
and then like use MAX_N + 1

File: solutions/cses-1132.mdx [hunk 13]
```diff
@@ -71,16 +76,101 @@ int main() {
...
+</Warning>
+	
+```java
+
+import java.util.*;
+import java.io.*;
+
+public class TreeDistancesI {
+	static ArrayList<Integer>[] adj;
+	static int [] firstMax = new int[200001];  // to store first-max length.
```
Comment (SansPapyrus683) [inline solutions/cses-1132.mdx:103]: same thing w/ MAX_N here

File: solutions/cses-1132.mdx (hunk 5 above)
Comment (SansPapyrus683) [reply solutions/cses-1132.mdx:107]: agreed

File: solutions/cses-1132.mdx [hunk 14]
```diff
@@ -2,60 +2,65 @@
...
+<LanguageSection>
+	
+<CPPSection>	
 ```cpp
-// source -> https://cses.fi/problemset/task/1132/
-
 #include<bits/stdc++.h>
 using namespace std;
 
 #define FIO ios::sync_with_stdio(0); cin.tie(0);
```
Comment (SansPapyrus683) [inline solutions/cses-1132.mdx:20]: remove this

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/cses-1132.mdx (hunk 4 above)
Comment (ar492) [reply solutions/cses-1132.mdx]: agree

File: solutions/cses-1132.mdx (hunk 12 above)
Comment (ar492) [reply solutions/cses-1132.mdx:25]: Up to Brad. a MAX_N would be nice, but this is fine too

**Review by ar492 — APPROVED** [review]

File: solutions/cses-1132.mdx (hunk 5 above)
Comment (SansPapyrus683) [reply solutions/cses-1132.mdx:105]: what why was this resolved lol

File: solutions/cses-1132.mdx (hunk 14 above)
Comment (SansPapyrus683) [reply solutions/cses-1132.mdx]: not addressed

Comment (brad182) [reply solutions/cses-1132.mdx]: ? I thought i did, its not there

File: solutions/cses-1132.mdx (hunk 5 above)
Comment (brad182) [reply solutions/cses-1132.mdx:105]: i'm not sure what you're seeing but when I view the code there's comments on top of each dfs function describing what it does

File: solutions/cses-1132.mdx [hunk 15]
```diff
@@ -71,16 +73,102 @@ int main() {
...
+			if (firstMax[x] + 1 > firstMax[v]) {
+				secondMax[v] = firstMax[v];
+				firstMax[v] = firstMax[x] + 1;
+				c[v] = x;
+			} else if (firstMax[x] + 1 > secondMax[v]) {
+				secondMax[v] = firstMax[x] + 1;
+			}
+		}
+	}
+
```
Comment (SansPapyrus683) [inline solutions/cses-1132.mdx:120]: still no comment here (java dfs2)

**Review by SansPapyrus683 — APPROVED** [review]
you know what sure

**Review by nathangong — APPROVED** [review]

**Pull request merged**